pymc3/distributions/simulator.py

```diff
@@ -115,14 +114,18 @@ def random(self, point=None, size=None):
         else:
             return np.array([self.function(*params) for _ in range(size)])
 
+
     def _str_repr(self, name=None, dist=None, formatting="plain"):
         if dist is None:
             dist = self
         name = name
         function = dist.function.__name__
         params = ", ".join([var.name for var in dist.params])
         sum_stat = self.sum_stat.__name__ if hasattr(self.sum_stat, "__call__") else self.sum_stat
-        distance = self.distance.__name__
+        if hasattr(self.distance, "__name__"):
+            distance = self.distance.__name__
+        else:
+            distance = self.distance.__class__.__name__
```

Comment:
Just a little suggestion to simplify:
```suggestion
        distance = getattr(self.distance, "__name__", self.distance.__class__.__name__)
```